utils/s3/credentials_providers/aws_credentials_provider_chain.cc

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "aws_credentials_provider_chain.hh"
+#include "utils/log.hh"
+#include <seastar/core/coroutine.hh>
+
+namespace aws {
+static logging::logger cpc_logger("cred_provider_chain");
+
+seastar::future<s3::aws_credentials> aws_credentials_provider_chain::get_aws_credentials() {
+    for (const auto& provider : providers) {
+        try {
+            auto creds = co_await provider->get_aws_credentials();
+            if (creds) {
+                cpc_logger.info("AWS credentials were successfully retrieved by {}.", provider->get_name());
+                co_return creds;
+            }
+            cpc_logger.warn("Retrieving AWS credentials by credentials provider {} failed.", provider->get_name());
```

Comment:
Prod nodes will be full of such warning, since env and config file providers are first in the list and (according to PR comment) should only provide creds for tests

Reply:
right, but the burst will happen once in 6-12 hours. Actually I can set it to debug level and if everything fails, we will fail later when trying to use empty creds (when all providers fail) or succeed. Is this the way we want to proceed?